### PR TITLE
[fixed] unused variable, standalone running, import doctest module

### PR DESCRIPTION
**Pull request by slowy07**
### **Describe your change:**
fixed unused variable, and using ``if __name__ == "__main__"``, and imported doctest module for following style code

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: matrix/spiral_print.py [hunk 1]
```diff
@@ -10,8 +10,8 @@
```
Comment (cclauss) [inline matrix/spiral_print.py:10]:
```suggestion
from collections.abc import Iterable
```
This covers `dict`, `list`, `set`, `tuple`, etc. so we can handle (almost) anything the caller sends our way.

File: matrix/spiral_print.py [hunk 2]
```diff
@@ -10,8 +10,8 @@
 
 def checkMatrix(a):
     # must be
-    if type(a) == list and len(a) > 0:
-        if type(a[0]) == list:
+    if type(a) == tuple and len(a) > 0:
+        if type(a[0]) == tuple:
```
Comment (cclauss) [inline matrix/spiral_print.py:14]:
```suggestion
    if a and if isinstance(a, Iterable):
```
PEP8 explains that we should not directly compare types.

File: matrix/spiral_print.py [hunk 3]
```diff
@@ -33,7 +33,7 @@ def spiralPrint(a):
     if checkMatrix(a) and len(a) > 0:
 
         matRow = len(a)
-        if type(a[0]) == list:
+        if type(a[0]) == tuple:
```
Comment (cclauss) [inline matrix/spiral_print.py:36]:
```suggestion
        if isinstance(a[0], Iterable):
```

**Review by cclauss — CHANGES_REQUESTED** [review]

File: matrix/spiral_print.py [hunk 4]
```diff
@@ -10,8 +10,7 @@
 from collections.abc import Iterable
 def checkMatrix(a):
```
Comment (cclauss) [inline matrix/spiral_print.py:11]:
```suggestion


def checkMatrix(a):
```

Comment (slowy07) [reply matrix/spiral_print.py:13]: solved on [8fc1402](https://github.com/TheAlgorithms/Python/pull/4673/commits/8fc14026089cf6d8237b88ecb0c409a7617ef6e7)

**Review by cclauss — APPROVED** [review]

**Pull request merged**